lib/knackhq/client/recursive_symbolize_keys.rb

```diff
@@ -0,0 +1,21 @@
+# Recursively symbolizes keys in hash
```

Comment:
Most gems I have seen do not have most of their code in the main lib/gem_name.rb file but have the functionality in a support folder: https://github.com/bbatsov/rubocop/blob/master/lib/rubocop.rb

I can add some comments to clarify I suppose.